### PR TITLE
CheatSheets: Assign some custom categories

### PR DESCRIPTION
**Pull request by GuiltyDolphin**
@zachthompson This should act as a decent trail of the new features. For example, we had quite a few cheat sheets for games that displayed 'cheats'; for those I've assigned the 'cheats' category.

---
IA Page: https://duck.co/ia/view/cheat_sheets
Maintainer: @zachthompson

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: share/goodie/cheat_sheets/triggers.yaml [hunk 1]
```diff
@@ -183,4 +183,28 @@ template_map:
...
 # much as possible to ensure it is as easy as possible to locate triggers.
-# custom_triggers:
+custom_triggers:
+  age_of_mythology_cheat_sheet:
+    additional_categories:
+      - "cheats"
+  boolean_algebra_cheat_sheet:
+    additional_categories:
+      - "math"
+  gta_san_andreas_cheat_sheet:
```
Comment (zachthompson) [inline share/goodie/cheat_sheets/triggers.yaml]: Let's not do it with something so common, e.g. game commands/cheats.  We don't want every game cheat sheet having to do this.  What we really want is for them to be in a "games" type but there is no template.  Maybe our first limitation with associating it with template_type.

Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/triggers.yaml]: @zachthompson Hmm... When we move on to a UI for cheat sheets I'm guessing they could just specify things like categories?

Wait a mo'... Could we base it off the categories metadata on duck.co?

Comment (zachthompson) [inline share/goodie/cheat_sheets/triggers.yaml]: That's an idea.

Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/triggers.yaml]: @zachthompson  For example, https://duck.co/ia/view/gta_san_andreas_cheat_sheet has 'Games' as a category. We could dynamically adapt triggers solely based on that.

Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/triggers.yaml]: @zachthompson Okay, I've got a rough version using `DDG::Meta::Data` which lets me grab the topics. I think perhaps a map from topics->categories in the `triggers.yaml` would suffice for deciding which categories to apply to which topics?

Comment (zachthompson) [inline share/goodie/cheat_sheets/triggers.yaml]: That would be the idea.  We'd want to think through whether the topics offered are sufficient to determine how they should trigger, what to do when multiple topics are specified, etc.

Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/triggers.yaml]: I was thinking just replace `additional_categories` with auto-generated `additional_categories`. I.e, *we* assign additional categories based on topics, and not all topics warrant a custom category etc...

Comment (zachthompson) [inline share/goodie/cheat_sheets/triggers.yaml]: Yep, I get that.  Just wondering if something like a topic of "Games" is going to be the right indicator for which triggers to apply.

Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/triggers.yaml]: @zachthompson Maybe the topics in conjunction with template? That or other topics.

I think a bit of over-triggering is fine? But yeah - I'll see if I can find anything that helps keep it more relevant.

(Of course, as you mention, we could have multiple topics (together) map to a category, which might help in some circumstances).

Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/triggers.yaml]: I could do a quick search of the description for some (e.g., search for 'cheat codes'), but I'm not sure how *'hacky'* we want to get.

Comment (zachthompson) [inline share/goodie/cheat_sheets/triggers.yaml]: Let's move the discussion about using topics to an issue.

Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/triggers.yaml]: @zachthompson Sure.

Comment (zachthompson) [inline share/goodie/cheat_sheets/triggers.yaml]: Let's close this for now until we know if topics can replace additional_categories? I don't want folks to start down this path and us having to redo it.

Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/triggers.yaml]: @zachthompson Sure.

**Pull request closed without merging**